cranelift/codegen/src/ir/memflags.rs

```diff
@@ -11,9 +11,20 @@ enum FlagBit {
     Readonly,
     LittleEndian,
     BigEndian,
+    /// Accesses the "heap" part of abstract state. Used for alias
+    /// analysis. Mutually exclusive with "table" and "vmctx".
+    Heap,
+    /// Accesses the "table" part of abstract state. Used for alias
+    /// analysis. Mutually exclusive with "heap" and "vmctx".
+    Table,
```

Comment:
Slightly unfortunate that we can't easily differentiate between two different memories or two different tables. I don't think it will matter too much in practice, but unfortunate none the less.

File a follow up issue for having an alias analysis bit for each memory/table instead of lumping all memories and all tables together?

Reply:
Sure! There are efficiency reasons why this is not trivial -- bits in a bitset (`MemFlags` is a `u8`, but even with a `u32` or `u64` that's still a small fixed number of heaps/tables), and the vector of last-stores (does it scale arbitrarily?). But maybe there's a hybrid approach (small number of privileged heaps/tables) or maybe we can find a way to scale it well enough.

Filed #4166.